### ISSUE TITLE
Limitation in the number of times you can @extend a placeholder selector?

### ISSUE THREAD

**Issue by gabrielecirulli** [sass/sass#1362]
I use an icon font to render icons in my project, and I use a placeholder selector to hold the basic attributes for all icons:

``` scss
%base-icon {
  font-family: "Icomoon"; // Font generated using Icomoon
  speak: none;
  font-style: normal;
  font-weight: normal;
  font-variant: normal;
  text-transform: none;

  // Better Font Rendering
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}
```

I then use a mixin called `icon` to include it in `:before` and `:after` pseudo-elements wherever I need icons to show up:

``` scss
@mixin icon($name, $family: false) {
  $glyph: map-get($icon-names, $name);
  content: $glyph;

  @if $family == true {
    @extend %base-icon;
  }
}
```

When `$family` is `true`, it'll extend `%base-icon`.

This morning I encountered a problem where after adding another icon in a pseudo-element, a few of them stop appearing (I see a small rectangle in place of the icon), as if SASS stopped outputting the selectors at all (I haven't had the time to investigate this aspect though, and I'll do it in a bit).

I pinpointed the amount of times you can extend to 7, after which a few of the icons will break. 

As a temporary solution I moved the contents of my placeholder selector to a mixin and then created multiple placeholder selectors that will be used for batches of 7 icons: when you include the 8th, it'll start extending the next placeholder selector:

``` scss
$base-icon-extend-counter: 1;
$base-icon-batch-size: 10;

@mixin icon($name, $family: false) {
  $glyph: map-get($icon-names, $name);
  content: $glyph;

  @if $family == true {
    $base-icon-batch-number: ceil($base-icon-extend-counter / $base-icon-batch-size);
    @extend %base-icon#{$base-icon-batch-number};

    $base-icon-extend-counter: $base-icon-extend-counter + 1;
  }
}
```

This solution works, but I'm not very happy with it.

Is there a known bug around this issue? I couldn't find anything on the internet regarding this.

**Comment by gabrielecirulli** [comment]
Well, I just figured it out. My last `&:before { ... }` somehow had a space with the ASCII code 160 before the curly braces, instead of the usual 32 (I have no idea how that happened, and it's been happening quite often to me lately using Sublime Text (e.g. sometimes just rewriting a line will fix the issue).

Anyway, that single "fake" space (it's actually an `&nbsp;`) was breaking the entire chain of extends since SASS just outputs a comma-separated list of all the selectors where I used `@extend`. That's also why I was seeing the odd limitation of 7 selector: adding the 8th would break the entire thing.
